components/tidb_query/src/codec/convert.rs

```diff
@@ -642,7 +643,8 @@ fn round_int_str(num_next_dot: char, s: &str) -> Cow<'_, str> {
     match s.rfind(|c| c != '9' && c != '+' && c != '-') {
         Some(idx) => {
             int_str.push_str(&s[..idx]);
-            let next_char = char::from_u32(s.chars().nth(idx).unwrap() as u32 + 1).unwrap();
+            // because the `s` must be valid int_str, so it is ok to do this.
+            let next_char = (s.as_bytes()[idx] + 1) as char;
             int_str.push(next_char);
             let zero_count = s.len() - (idx + 1);
             if zero_count > 0 {
```

Comment:
prefer to use `std::repeat` to appending zeros.

Reply:
```rust
#[cfg(test)]
mod tests {
    use super::*;
    use test::Bencher;

    #[bench]
    fn bf_extend(b: &mut test::Bencher) {
        let mut len = 0;
        b.iter(|| {
            let mut int_str = String::with_capacity(878);
            let zero_count = 878;
            int_str.extend((0..zero_count).map(|_| '0'));
            len += int_str.len();
        });
        println!("{}", len);
    }

    #[bench]
    fn bf_rep(b: &mut test::Bencher) {
        let mut len = 0;
        b.iter(|| {
            let mut int_str = String::with_capacity(878);
            int_str.extend(std::iter::repeat('0').take(879));
            len += int_str.len();
        });
        println!("{}", len);
    }

    #[bench]
    fn bf_tri(b: &mut test::Bencher) {
        let mut len = 0;
        b.iter(|| {
            let mut int_str = String::with_capacity(878);
            for i in 0..878 {
                int_str.push('0');
            }
            len += int_str.len();
        });
        println!("{}", len);
    }
}
```

```

running 3 tests
test bm::tests::bf_extend ... bench:       1,215 ns/iter (+/- 185)
test bm::tests::bf_rep    ... bench:       1,313 ns/iter (+/- 194)
test bm::tests::bf_tri    ... bench:         705 ns/iter (+/- 70)
````